verifier/src/main/java/com/aws/jverify/verifier/JavaToDafnyCompiler.java

```diff
@@ -417,16 +394,23 @@ private Statement translateStatement(JCTree.JCStatement statement) {
                 List<AssignmentRhs> rhss = List.of(new ExprRhs(e.getOrigin(), null, e));
                 initializer = new AssignStatement(e.getOrigin(), null, lhss, rhss, false);
             }
-            
+
             return new VarDeclStmt(origin, null, List.of(localVariable), initializer);
         } else if (statement instanceof JCTree.JCWhileLoop whileLoop) {
 
+            List<AttributedExpression> preconditions = new ArrayList<>();
+            List<AttributedExpression> postconditions = new ArrayList<>();
             List<AttributedExpression> invariants = new ArrayList<>();
             List<Expression> decreases = new ArrayList<>();
             List<FrameExpression> modifies = new ArrayList<>();
-            Function<JCTree.JCStatement, Boolean> function =
-                    (JCTree.JCStatement s) -> findWhileHeaderStatement(s, invariants, decreases, modifies);
-            var statements = processStatementWithHeader(whileLoop.body, function);
+            List<Name> returnNames = new ArrayList<>();
+            
+            var statements = processStatementWithHeader(whileLoop.body, preconditions, postconditions,
+                    returnNames, invariants, decreases, modifies);
+            
+            checkEmptyExpressions(preconditions, "preconditions", "loop");
+            checkEmptyExpressions(preconditions, "postconditions", "loop");
```

Comment:
I think you meant this?

```suggestion
            checkEmptyExpressions(preconditions, "preconditions", "loop");
            checkEmptyExpressions(postconditions, "postconditions", "loop");
```